financial-templates-lib/helpers/GasEstimator.js

```diff
@@ -65,15 +67,33 @@ class GasEstimator {
         let price = json.fast;
         return price;
       } else {
-        throw new Error("bad json response");
+        throw new Error("Etherchain API: bad json response");
       }
     } catch (error) {
-      this.logger.error({
+      this.logger.debug({
         at: "GasEstimator",
-        message: "client polling error🚨",
+        message: "client polling error, trying backup API🚨",
         error: typeof error === "string" ? new Error(error) : error
       });
 
+      // Try backup API.
+      try {
+        const responseBackup = await fetch(backupUrl);
+        const jsonBackup = await responseBackup.json();
+        if (jsonBackup.SafeGasPrice) {
+          return jsonBackup.fast;
```

Comment:
Can you document the structure of this JSON? Is it possible for `SafeGasPrice` to exist but `fast` to not exist? Just want to make sure we don't return `undefined`.

Reply:
Sorry, noticed this after my initial approval.

Reply:
Ah this looks like a typo it should read `jsonBackup.SafeGasPrice`

Reply:
On an unrelated note, do they just have safe? Or do they have fast too?

Reply:
Fixed

Reply:
THey have something called `ProposeGasPrice` which appears faster we can use that.  It looks equivalent to the `FastestGasPrice` for etherchain.